src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -399,10 +401,108 @@ void generateDeclarations() {
     }
   }
 
+  private boolean isGoogCall(Node node) {
+    if (!node.isCall()) {
+      return false;
+    }
+
+    if (node.getFirstChild() == null || !node.getFirstChild().isGetProp()) {
+      return false;
+    }
+
+    Node getPropNode = node.getFirstChild();
+
+    if (getPropNode.getFirstChild() == null || !getPropNode.getFirstChild().isName()) {
+      return false;
+    }
+
+    return getPropNode.getFirstChild().getString().equals("goog");
+  }
+
+  private boolean isGoogRequire(Node node) {
+    if (!isGoogCall(node)) {
+      return false;
+    }
+
+    if (node.getFirstChild() == null) {
+      return false;
+    }
+
+    Node getPropNode = node.getFirstChild();
+
+    if (getPropNode.getChildAtIndex(1) == null || !getPropNode.getChildAtIndex(1).isString()) {
+      return false;
+    }
+    return getPropNode.getChildAtIndex(1).getString().equals("require");
+  }
+
+  private boolean isGoogModule(Node node) {
```

Comment:
It's unclear from the name whether this checks if the Node is a source file that's a goog.module, or whether it checks if this is the goog.module call. Maybe "call" should appear in the name?